pkg/changelog/fake_http_impl.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package changelog
+
+import (
+	"io"
+	"os"
+	"strings"
+	"text/template"
+
+	"github.com/blang/semver"
+	"github.com/yuin/goldmark/parser"
+
+	"k8s.io/release/pkg/git"
+	"k8s.io/release/pkg/github"
+	"k8s.io/release/pkg/notes"
+	"k8s.io/release/pkg/notes/document"
+	"k8s.io/release/pkg/notes/options"
+)
+
+type FakeHttpImpl struct {
+	fallbackImpl impl
+}
+
+func NewFakeHttpImpl() *FakeHttpImpl {
+	return NewFakeHttpImplWithFallback(&defaultImpl{})
+}
+
+func NewFakeHttpImplWithFallback(fallback impl) *FakeHttpImpl {
+	return &FakeHttpImpl{
+		fallbackImpl: fallback,
+	}
+}
+
+func (i *FakeHttpImpl) GetURLResponse(url string, trim bool) (string, error) {
+	var result string
+	var err error
+
+	switch {
+	case strings.Contains(url, "release-notes-draft.md"):
+		oldUrl := "https://raw.githubusercontent.com/kubernetes/sig-release/master/releases/release-1.17/release-notes-draft.md"
+		result, err = i.fallbackImpl.GetURLResponse(oldUrl, trim)
+	default:
+		result, err = i.fallbackImpl.GetURLResponse(url, trim)
+	}
+
+	return result, err
+}
```

Comment:
to be honest I'm not very happy with how this is done but I don't know how else I should approach it without doing a major change on the `impl` interface structure the way it is today. thoughts? @hasheddan

Reply:
@wilsonehusin can we not just override this is the existing generated fake implementation? https://github.com/kubernetes/release/blob/5b258d8961ca446353cca43e6f5f6653b9a3eefa/pkg/changelog/changelogfakes/fake_impl.go#L941

Reply:
@hasheddan I've never worked with generated fakes before -- is the expectation to maintain it as if we write the code ourselves or is it to keep it the same as generated?

Reply:
@wilsonehusin sorry I think my suggestion may have been presented in an unclear manner. The purpose of the fakes is to be able to plug in your own logic for any of the methods. I am not suggesting literally modifying the fake implementation, but rather instantiating it in the tests with the proper override. So in this case, you would create a `&FakeImpl{}` with `GetURLResponseStub` set to your logic. Does that make sense? Happy to also pair or show an example as well :)

https://github.com/kubernetes/release/blob/5b258d8961ca446353cca43e6f5f6653b9a3eefa/pkg/changelog/changelogfakes/fake_impl.go#L154

Reply:
oooooohhh I didn't realize the methods were appended with `Stub`, I'll give it another try and will let you know! thanks Dan :)

Reply:
I'm going down this route and it seems like I'll have to stub almost every method in `impl` to use the `defaultImpl` for tests to pass, or quite literally mocking every `impl` which... seems to be bigger than the scope of this PR?

is there a way to tell the fakes to fallback to `defaultImpl`?

Reply:
ping following up @hasheddan 😄 I'm trying to find a compromise with the least that I can do to get this merged while also heading towards the right direction for refactoring

Reply:
it just 💡 in my head so i'm going to mark this one as resolved, let's discuss on the most recent changelog if needed